internal/system/agent/application/executor/metrics.go

```diff
@@ -45,7 +45,7 @@ func (m *metrics) Get(_ context.Context, services []string) ([]interface{}, erro
 		go func(serviceName string) {
 			defer wg.Done()
 
-			res, err := m.executor(m.executorPath, serviceName, "metrics")
+			res, err := m.executor(m.executorPath, addEdgexPrefix(serviceName), "metrics")
```

Comment:
Do we expect these service to just be the `edgex` service or can they be 3rd party service like `redis` and `kong` . If includes 3rd party, we will need to update the docker compose to name all 3rd party containers with `edgex-` prefix. Currently it is inconsistent for 3rd party container names. We have `edgex-redis`, `edgex-core-consul`, `edgex-vault` and then have `kong` and `kong-db`  and then `edgex-app-service-configurable-rules` which doesn't follow the service key approach....

Reply:
Added these issues:
https://github.com/edgexfoundry/edgex-compose/issues/126
https://github.com/edgexfoundry/edgex-taf/issues/464

Reply:
> Do we expect these service to just be the `edgex` service or can they be 3rd party service like `redis` and `kong` . If includes 3rd party, we will need to update the docker compose to name all 3rd party containers with `edgex-` prefix. Currently it is inconsistent for 3rd party container names. We have `edgex-redis`, `edgex-core-consul`, `edgex-vault` and then have `kong` and `kong-db` and then `edgex-app-service-configurable-rules` which doesn't follow the service key approach....

Yep we need to make a decision here (or maybe discuss in coreWG?) 
Although I kinda want this to be a more general API (you can operate 3rd party services, or even custom one), considering the naming inconsistent it seems impossible.
I'll be in favor to make this edgex service only. For device services we can directly add `edgex-` prefix. We might need to update app service and security services in either core-contracts or edgex-compose.

Reply:
> considering the naming inconsistent it seems impossible.

Since we have total control in the compose files we can fix the naming inconsistency.

Reply:
@hahattan, container/hostnames have been updated so all services running in docker have the `edgex-`, so your code can assume to just add the prefix to any service name it receives.